airbyte-cdk/python/airbyte_cdk/entrypoint.py

```diff
@@ -144,7 +144,14 @@ def check(self, source_spec: ConnectorSpecification, config: TConfig) -> Iterabl
                 yield connection_status
                 return
 
-        check_result = self.source.check(self.logger, config)
+        try:
+            check_result = self.source.check(self.logger, config)
+        except AirbyteTracedException as traced_exc:
```

Comment:
Even after https://airbytehq-team.slack.com/archives/C03AS1GAQV6/p1725902027876519 and AJ's comment, I feel like I'm still missing context: Why do we only catch AirbyteTracedException here? If there is a `ValueError` for example, it feels like we would only return the TRACE message and not the connection status

Reply:
@maxi297 - I think we want to distinguish two things:
1. The connector is broken and behaving incorrectly due to bad code. This requires an engineer to fix.
2. The connector is fine but the connector config provided by the user is incorrect, or the creds they provided are stale, etc...

In PyAirbyte, we have an explicit exception type called "PyAirbyteInternalError" which is distinct from user errors - and I think that's the spirit here as well. I'd like CAT to fail, for instance, if there's an internal (uncaugh/unhandled) exception, and on the other side, I'd like to continually move towards more friendly remediation recommendations for _handled_ and expected exception types: password expired, invalid paths, etc. 

Wdyt?

Reply:
That said, I'm certainly not against catching _all_ exceptions, so long as we message them differently, and hopefully also leave space for CAT to catch them as not-yet-handled.

Reply:
I'm not sure this code is good enough then. I think the platform expects the docker container to fail with error code != 0 if there is an issue (at least, this is true on read where we added [this](https://github.com/airbytehq/airbyte/blob/bb1833cbf558b963bccc1efa07408c50685b8c8b/airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py#L174-L177)). By catching all `AirbyteTracedException` here and not raising the error in case of system_error, I fear we might be losing some alerting

Reply:
@benmoriceau as move OC - can you help clarify the platform's expectations with regards to the check operation?

What exactly do we need to properly surface these to the platform?
- a config error
- a system error

Reply:
As part of @girarda 's question, can you add details about how it impacts alerting?

Reply:
[From slack](https://airbytehq-team.slack.com/archives/C03AS1GAQV6/p1725908284013149?thread_ts=1725902027.876519&cid=C03AS1GAQV6):
- system error: we need to raise the exception to exit with a non-zero code
- config error: we need to return the AirbyteTraceMessage + failed connection status